src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1863,6 +1862,10 @@ internal async Task Reset(bool async, CancellationToken cancellationToken = defa
                 }
             }
 
+            // TODO: there is a bug, when we're forcefuly closing the connection while inside of the transaction block
+            // and after that attempt to dispose the transaction
+            Transaction?.UnbindIfNecessary();
```

Comment:
This can move up into the TransactionStatus switch's Idle case, no? In the other cases, we call ClearTransaction so we know the transaction is disposed.

Reply:
This doesn't guarantee anything - we still may dispose the transaction after another connection taket the connector from the pool and resets the transaction.

Reply:
You're right, but in that case your current PR has a problem, no? Because UnbindIfNecessary won't do anything, since ClearTransaction has already set IsDisposed=true.

So UnbindIfNecessary needs to happen before we call ClearTransaction?

Reply:
> You're right, but in that case your current PR has a problem, no? Because UnbindIfNecessary won't do anything, since ClearTransaction has already set IsDisposed=true.
> 
> So UnbindIfNecessary needs to happen before we call ClearTransaction?

That is why there is a TODO where - I'm really not sure. I would rather definitely fix something (and maybe leave something still broken), than break something else. If you do not see a problem there, than ok.